jobs/microk8s/clusterapi/release.sh

```diff
@@ -0,0 +1,158 @@
+#!/bin/bash
+
+set -xe
+
+#################################################################
+### STAGE 0: Description
+
+## Description:
+## - Fetches Git repositories from bootstrap and control plane providers, builds controllers,
+##   spins up a test cluster on AWS and runs a few basic checks, then tears everything down
+##   and releases.
+## - Release means pushing Docker images to DockerHub and tags to GitHub repositories.
+##   There is a GitHub action configured on the repositories to create a release on tag push.
+
+## Assumptions
+## - Runs in an LXD environment where Docker and MicroK8s can be installed
+
+## Limitations
+## - Currently does nothing to manage cleaning up stale AWS resources in case of failure
+
+## Requirements
+## - docker
+## - git
+## - build-essential
+
+## Credentials
+## - AWS (to run tests)
+## - Git (to push tags to GitHub repositories)
+## - DockerHub (to push images)
+
+## Configuration
+BOOTSTRAP_PROVIDER_CHECKOUT="${BOOTSTRAP_PROVIDER_CHECKOUT:-main}"              # commit to checkout from bootstrap provider
+CONTROL_PLANE_PROVIDER_CHECKOUT="${CONTROL_PLANE_PROVIDER_CHECKOUT:-main}"      # commit to checkout from control plane provider
+RELEASE_TAG="${RELEASE_TAG:-0.3.0-rc1}"                                         # release tag that will be pushed on success
+
+#################################################################
+### STAGE 1: Build Docker images and release manifests from the checked out source code
+
+git clone https://github.com/canonical/cluster-api-bootstrap-provider-microk8s bootstrap -b "${BOOTSTRAP_PROVIDER_CHECKOUT}"
+git clone https://github.com/canonical/cluster-api-control-plane-provider-microk8s control-plane -b "${CONTROL_PLANE_PROVIDER_CHECKOUT}"
+(
+    cd bootstrap
+    docker build -t cdkbot/capi-bootstrap-provider-microk8s-dev:${RELEASE_TAG} .
```

Comment:
Should this "-dev" be better be be a ${RELEASE_TAG}-rc ? I mean instead of pushing a different image `capi-bootstrap-provider-microk8s-dev` shouldn't we push an `-rc` tag or an `under-testing` tag?

Reply:
the purpose is to avoid clattering the image with temporary dev tags; feel free to change as needed.